### PR TITLE
INT-2541 Add shutdown support for NXRM

### PR DESCRIPTION
**Pull request by adrianpowell**
Jira: https://issues.sonatype.org/browse/INT-2541

Added support for users to add hooks in startup to remove any stale lock files, if desired. It also adds support for an increased grace period as described in the sonatype help docs for docker images.

Also boyscoutted an improvement to the RHCC pull secret so that it can be easily added to the values file rather than being stored separately. This will allow `helm del --purge nxrm` during testing.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 ﻿OpenShift/helm/nexus-iq/myvalues.yaml
+.idea/*
```
Comment (jflinchbaugh) [inline .gitignore]: could this just be `.idea` so it's not tempted to even get the directory?

File: OpenShift/helm/nexus-repository-manager/values.yaml [hunk 2]
```diff
@@ -149,6 +149,9 @@ deployment:
   # # Uncomment and modify this to run a command after starting the nexus container.
   postStart:
     command:    # '["/bin/sh", "-c", "ls"]'
+  preStart:
+    command:    # '["/bin/rm", "-f", "/tmp/i4jdaemon__opt_nexus_nexus-3.20.0-04_bin_nexus"]'
```
Comment (jflinchbaugh) [inline OpenShift/helm/nexus-repository-manager/values.yaml]: what's this path? is there a better example path we could provide?

Comment (adrianpowell) [reply OpenShift/helm/nexus-repository-manager/values.yaml]: That was the path used in the helpdoc for the PID lockfile. But it's ugly and untested so I'll make it a little clearer.

File: .gitignore (hunk 1 above)
Comment (adrianpowell) [reply .gitignore]: I used a tool's defaults but i can make this better  👍

Comment (collinpeters) [reply .gitignore]: FWIW Git doesn't track directories

**Pull request merged**